### PR TITLE
Migrate to typescript

### PR DESCRIPTION
**Pull request by szeck87**
#1

### CODE REVIEW

**Comment by szeck87** [comment]
Once https://github.com/Unibeautify/unibeautify/pull/18 is merged in Unibeautify, it will give this error when running tests:

```
  TypeError: Cannot convert undefined or null to object
    Function.keys (<anonymous>)
    _.find (/path/to/project//beautifier-prettydiff/node_modules/unibeautify/src/beautifier.ts:294:92)
    baseFindIndex (/path/to/project//beautifier-prettydiff/node_modules/lodash/lodash.js:823:11)
    findIndex (/path/to/project//beautifier-prettydiff/node_modules/lodash/lodash.js:7275:14)
    Function.find (/path/to/project//beautifier-prettydiff/node_modules/lodash/lodash.js:5098:21)
    Unibeautify.getBeautifierForLanguage (/path/to/project//beautifier-prettydiff/node_modules/unibeautify/src/beautifier.ts:294:14)
    Unibeautify.beautify (/path/to/project//beautifier-prettydiff/node_modules/unibeautify/src/beautifier.ts:204:53)
    Test.fn (index.js:29:22)
    Array.map (<anonymous>)
    process._tickCallback (internal/process/next_tick.js:152:19)
```

**Comment by Glavin001** [comment]
Can you add a `.unibeautifyrc.yml` file?

```yaml
---
_:
  indent_char: " "
  indent_size: 2
TypeScript:
  indent_char: " "
  indent_size: 2
JavaScript:
  indent_char: " "
  indent_size: 2
CSS:
  indent_char: " "
  indent_size: 2
HTML:
  indent_char: " "
  indent_size: 2
JSON:
  indent_char: " "
  indent_size: 2
```

^ And feel free to change config above.

File: .unibeautifyrc.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+JavaScript:
+  indent_char: " "
+  indent_size: 2
+CSS:
+  indent_char: " "
+  indent_size: 2
+HTML:
+  indent_char: " "
+  indent_size: 2
+JSON:
```
Comment (Glavin001) [inline .unibeautifyrc.yml]: Let's remove the `JSON` formatting. It differs from `npm`'s `package.json` file format.

Just delete the whole `JSON` property, including sub-properties. Thanks!

Comment (szeck87) [reply .unibeautifyrc.yml]: Done.

**Comment by szeck87** [comment]
OK, I think this is stable.

File: src/index.ts [hunk 2]
```diff
@@ -79,7 +79,7 @@ export const beautifier: Beautifier = {
           }
         }
       ],
-      ternaryline: "preserve_ternary_lines"
+      ternaryline: "multiline_ternary"
```
Comment (Glavin001) [inline src/index.ts]: I am fairly certain this is a bug.

`multiline_ternary` can be `always`, `always-multiline`, or `never`. 
I think `ternaryline` expects `boolean`. So will need a transform option handler.

File: src/index.ts [hunk 3]
```diff
@@ -81,13 +81,14 @@ export const beautifier: Beautifier = {
...
-          } else {
-            return undefined;
+        (options): boolean | undefined => {
+          switch (options.multiline_ternary) {
+            case "always":
+              return false;
+            case "never":
+              return true;
+            default:
+              return undefined;
```
Comment (Glavin001) [inline src/index.ts]: I've been returning `undefined` with the expectation the base beautifier will simply NOT do anything with regards to this option. Hopefully Pretty Diff respects this, too. Prettier seemed to be OK with `undefined` option values.

Comment (szeck87) [reply src/index.ts]: If it doesn't, I think we should return true for this particular option since that is prettydiff's default.

**Pull request merged**